--- origsrc/Botan-1.10.5/configure.py	2013-03-02 15:19:36.000000000 -0600
+++ src/Botan-1.10.5/configure.py	2014-02-02 16:05:37.897195900 -0600
@@ -922,7 +922,7 @@ class OsInfo(object):
         self.build_shared = (True if self.build_shared == 'yes' else False)
 
     def ranlib_command(self):
-        return ('ranlib' if self.ar_needs_ranlib else 'true')
+        return ('$(CROSS)ranlib' if self.ar_needs_ranlib else 'true')
 
     def defines(self):
         return ['TARGET_OS_IS_%s' % (self.basename.upper())] + \
--- origsrc/Botan-1.10.5/src/build-data/cc/gcc.txt	2013-03-02 15:19:37.000000000 -0600
+++ src/Botan-1.10.5/src/build-data/cc/gcc.txt	2014-02-02 16:11:36.240691900 -0600
@@ -1,6 +1,6 @@
 macro_name GCC
 
-binary_name g++
+binary_name "$(CROSS)g++"
 
 has_tr1 yes
 
@@ -17,7 +17,7 @@ maintainer_warning_flags "-Werror -Wall
 
 lib_opt_flags "-O3"
 check_opt_flags "-O2"
-shared_flags "-fPIC"
+shared_flags "-DPIC"
 debug_flags -g
 no_debug_flags "-finline-functions"
 
@@ -39,6 +39,7 @@ openbsd -> "$(CXX) -shared -fPIC"
 darwin  -> "$(CXX) -dynamiclib -fPIC -install_name $(LIBDIR)/$(SONAME)"
 hpux    -> "$(CXX) -shared -fPIC -Wl,+h,$(SONAME)"
 solaris -> "$(CXX) -shared -fPIC -Wl,-h,$(SONAME)"
+mingw   -> "$(CXX) -shared -Wl,--out-implib,$(SYMLINK)"
 </so_link_flags>
 
 <mach_opt>
--- origsrc/Botan-1.10.5/src/build-data/makefile/unix_shr.in	2013-03-02 15:19:37.000000000 -0600
+++ src/Botan-1.10.5/src/build-data/makefile/unix_shr.in	2014-02-02 16:27:27.363093000 -0600
@@ -57,10 +57,10 @@ LIBRARIES     = $(STATIC_LIB) $(SHARED_L
 LIBNAME       = %{lib_prefix}libbotan
 STATIC_LIB    = $(LIBNAME)-$(SERIES).a
 
-SONAME        = $(LIBNAME)-$(SERIES).%{so_suffix}.%{so_abi_rev}
-SHARED_LIB    = $(SONAME).%{version_patch}
+SONAME        = $(LIBNAME)-$(SERIES)-%{so_abi_rev}.%{so_suffix}
+SHARED_LIB    = $(SONAME)
 
-SYMLINK       = $(LIBNAME)-$(SERIES).%{so_suffix}
+SYMLINK       = $(LIBNAME)-$(SERIES).dll.a
 
 all: $(LIBRARIES)
 
@@ -80,8 +80,6 @@ $(STATIC_LIB): $(LIBOBJS)
 
 $(SHARED_LIB): $(LIBOBJS)
 	$(SO_LINK_CMD) $(LDFLAGS) $(LIBOBJS) -o $(SHARED_LIB) $(LINK_TO)
-	$(LN) $(SHARED_LIB) $(SONAME)
-	$(LN) $(SHARED_LIB) $(SYMLINK)
 
 # Fake Targets
 .PHONY = docs clean distclean install static shared
@@ -113,12 +111,10 @@ install: $(LIBRARIES) docs
 	 done
 
 	$(MKDIR_INSTALL) $(LIBDIR)
-	$(INSTALL_CMD_DATA) $(STATIC_LIB) $(LIBDIR)
-	$(INSTALL_CMD_EXEC) $(SHARED_LIB) $(LIBDIR)
-	$(CD) $(LIBDIR); $(LN) $(SHARED_LIB) $(SYMLINK)
-	$(CD) $(LIBDIR); $(LN) $(SHARED_LIB) $(SONAME)
-
 	$(MKDIR_INSTALL) $(BINDIR)
+	$(INSTALL_CMD_DATA) $(STATIC_LIB) $(LIBDIR)
+	$(INSTALL_CMD_EXEC) $(SHARED_LIB) $(BINDIR)
+	$(INSTALL_CMD_DATA) $(SYMLINK) $(LIBDIR)
 	$(INSTALL_CMD_EXEC) $(CONFIG_SCRIPT) $(BINDIR)
 
 	$(MKDIR_INSTALL) $(PKGCONF_DIR)
--- origsrc/Botan-1.10.5/src/build-data/os/mingw.txt	2013-03-02 15:19:37.000000000 -0600
+++ src/Botan-1.10.5/src/build-data/os/mingw.txt	2014-02-02 16:17:20.923406700 -0600
@@ -4,9 +4,7 @@ obj_suffix o
 so_suffix dll
 static_suffix a
 
-build_shared no
-
-ar_command "ar crs"
+ar_command "$(CROSS)ar crs"
 ar_needs_ranlib yes
 
 install_root /mingw
